File: src/main/java/it/giaquinto/springberry/http/HttpRequest.java
package it.giaquinto.springberry.http;

import org.springframework.web.bind.annotation.RequestMethod;

public enum HttpRequest {
    GET,
    HEAD,
    POST,
    PUT,
    DELETE,
    OPTIONS,
    TRACE,
    PATCH;


    public static RequestMethod get(final HttpRequest toConvert) {
        return switch (toConvert) {
            case GET -> RequestMethod.GET;
            case HEAD -> RequestMethod.HEAD;
            case POST -> RequestMethod.POST;
            case PUT -> RequestMethod.PUT;
            case DELETE -> RequestMethod.DELETE;
            case OPTIONS -> RequestMethod.OPTIONS;
            case TRACE -> RequestMethod.TRACE;
            case PATCH -> RequestMethod.PATCH;
        };
    }
}

File: src/main/java/it/giaquinto/springberry/model/http/HttpRequest.kt
package it.giaquinto.springberry.model.http

import org.springframework.web.bind.annotation.RequestMethod

enum class HttpRequest {
    GET, HEAD, POST, PUT, DELETE, OPTIONS, TRACE, PATCH;

    companion object {
        operator fun get(toConvert: HttpRequest): RequestMethod =
            when (toConvert) {
                GET -> RequestMethod.GET
                HEAD -> RequestMethod.HEAD
                POST -> RequestMethod.POST
                PUT -> RequestMethod.PUT
                DELETE -> RequestMethod.DELETE
                OPTIONS -> RequestMethod.OPTIONS
                TRACE -> RequestMethod.TRACE
                PATCH -> RequestMethod.PATCH
            }
    }
}